lib/nerves_motd/runtime.ex

```diff
@@ -43,4 +49,28 @@ defmodule NervesMOTD.Runtime.Target do
       |> tl
       |> Enum.map(&String.to_integer/1)
   end
+
+  @impl NervesMOTD.Runtime
+  def filesystem_stats(filename) when is_binary(filename) do
+    case System.cmd("df", ["-m", filename]) do
+      {df_results, 0} ->
+        [size_mb, used_mb, _, used_percent, _] =
+          df_results
+          |> String.split("\n")
+          |> tl
+          |> Enum.map(&String.split/1)
+          |> Enum.reject(&match?([], &1))
+          |> Enum.map(fn [_path | data] -> data end)
+          |> hd
+
+        %{
+          size_mb: elem(Integer.parse(size_mb), 0),
+          used_mb: elem(Integer.parse(used_mb), 0),
+          used_percent: elem(Integer.parse(used_percent), 0)
+        }
+
+      _ ->
+        raise "File not found"
+    end
```

Comment:
This requires changing the caller to deal with `{:ok, result}` and `:error`. It also handles a lot of potential parse errors. I could go with raising on parse errors, since they really wouldn't be expected. That might be another improvement, but I figured that I'd share this as a step in that direction.

```suggestion
    # Use df to determine filesystem statistics. df's output looks like:
    #
    #     Filesystem           1M-blocks      Used Available Use% Mounted on
    #     /dev/mmcblk0p4            1534       205      1329  13% /root

    with {df_results, 0} <- System.cmd("df", ["-m", filename]),
         [_title_row, results_row | _] <- String.split(df_results, "\n"),
         [_fs, size_mb_str, used_str, _avail, used_percent_str | _] <- String.split(results_row),
         {size_mb, ""} <- Integer.parse(size_mb_str),
         {used_mb, ""} <- Integer.parse(used_str),
         {used_percent, "%"} <- Integer.parse(used_percent_str) do
      {:ok,
       %{
         size_mb: size_mb,
         used_mb: used_mb,
         used_percent: used_percent
       }}
    else
      _ -> :error
    end
```

Reply:
👍 Your version would be better in identifying potential parse errors.

Reply:
I realized this pattern can be applied to `memory_usage/0`, I'll refactor it later in a separate PR.